Fix output extra comma

FengyunPan opened the pull request:
/# kubectl get service
NAME                     CLUSTER-IP       EXTERNAL-IP               PORT(S)          AGE
nginx-service-test       10.200.6.109     _**,80.11.12.10**_              8000:31637/TCP   6s
**(output a extra comma)**
except: 
nginx-service-test       10.200.6.109     _**80.11.12.10**_              8000:31637/TCP   6s

/# cat lb.yaml 
apiVersion: v1 
kind: Service 
metadata: 
  name: nginx-service-lb3
spec: 
  ports: 
    - port: 8000
      targetPort: 80 
      protocol: TCP 
  type: LoadBalancer
  selector: 
    name: nginx
  externalIPs: 
    - 80.11.12.10
    - 1.2.3.6
/# kubectl create -f lb.yaml

FengyunPan commented:
/release-note-none

deads2k commented on pkg/printers/internalversion/describe.go:
```diff
@@ -1918,7 +1918,9 @@ func describeService(service *api.Service, endpoints *api.Endpoints, events *api
 		w.Write(LEVEL_0, "Selector:\t%s\n", labels.FormatLabels(service.Spec.Selector))
 		w.Write(LEVEL_0, "Type:\t%s\n", service.Spec.Type)
-		w.Write(LEVEL_0, "IP:\t%s\n", service.Spec.ClusterIP)
+		if service.Spec.ClusterIP != "" {
```
Simplest is to have unconditional output.  If you want to conditionalize it, you'd need to relate it to the various service types and deal with exceptional cases.  I recommend leaving it as-is.

FengyunPan replied:
ok

deads2k commented on pkg/printers/internalversion/printers.go:
```diff
@@ -715,7 +715,12 @@ func getServiceExternalIP(svc *api.Service, wide bool) string {
 		lbIps := loadBalancerStatusStringer(svc.Status.LoadBalancer, wide)
 		if len(svc.Spec.ExternalIPs) > 0 {
-			result := append(strings.Split(lbIps, ","), svc.Spec.ExternalIPs...)
+			var result []string
```
seems like the old code does the same thing more compactly.

FengyunPan replied:
Thank for your review very much. The old code will add a incorrect comma into output. My change can fix it.

eparis replied:
Would it read better like:
```
results := []string{}
if len(lbIPs) > 0 {
    results = append(results, strings.Split(lbIPs, ",")
}
results = append(results, svc.Spec.ExternalIPs)
```
Performance wise it might be a little slower with the allocations and appends but it would read/review quite easily.

FengyunPan replied:
Nice, I update it soon, thank you very much.

deads2k commented on pkg/printers/internalversion/printers.go:
```diff
@@ -746,18 +751,26 @@ func printService(svc *api.Service, w io.Writer, options printers.PrintOptions)
 	namespace := svc.Namespace
 
 	internalIP := svc.Spec.ClusterIP
+	if internalIP == "" {
```
style for this project: `len(internalIP) == 0`

FengyunPan replied:
Ok, done.

Commit:
Display <none> for clusterIP and port when service is ExternalName

FengyunPan commented:
@deads2k It is hard to understande it, and I show you a test:
$cat test.go
func main(){
	ExternalIPs := []string{"10.0.0.6"}
	test := []string{}
	r := strings.Join(test, ",")
	result := append(strings.Split(r, ","), ExternalIPs...)
	res := strings.Join(result, ",")
	fmt.Printf("-------%s--------\n", res)
}  
$ go run test.go 
-------,10.0.0.6--------
         |
         |
 Extra comma

eparis commented:
@k8s-bot ok to test
/lgtm

eparis commented:
/lgtm

deads2k commented:
/approve no-issue

FengyunPan commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

Commit:
Fix output extra comma

When running 'kubectl get service', I get a extra comma in result:
'EXTERNAL-IP:,172.16.0.12', but except: 'EXTERNAL-IP:172.16.0.12'.

FengyunPan commented:
ping @deads2k 
ping @eparis

deads2k commented:
> @deads2k It is hard to understande it, and I show you a test:

thanks

/lgtm
